src/seedu/addressbook/Main.java

```diff
@@ -13,117 +13,136 @@
+	private AddressBook addressBook;
+
+	/** The list of person shown to the user most recently. */
+	private List<? extends ReadOnlyPerson> lastShownList = Collections.emptyList();
+
+	public static void main(String... launchArgs) {
+		new Main().run(launchArgs);
+	}
+
+	/** Runs the program until termination. */
+	public void run(String[] launchArgs) {
+		start(launchArgs);
+		runCommandLoopUntilExitCommand();
+		exit();
+	}
+
+	/**
+	 * Sets up the required objects, loads up the data from the storage file,
+	 * and prints the welcome message.
+	 *
+	 * @param launchArgs
+	 *            arguments supplied by the user at program launch
+	 *
+	 */
+	private void start(String[] launchArgs) {
+		try {
+			this.ui = new TextUi();
+			this.storage = initializeStorage(launchArgs);
+			this.addressBook = storage.load();
+			ui.showWelcomeMessage(VERSION, storage.getPath());
+
+		} catch (InvalidStorageFilePathException | StorageOperationException | FileDeletedException e) {
+			ui.showInitFailedMessage();
+			/*
+			 * ==============NOTE TO
+			 * STUDENTS=========================================================
+			 * ================ We are throwing a RuntimeException which is an
+			 * 'unchecked' exception. Unchecked exceptions do not need to be
+			 * declared in the method signature. The reason we are using an
+			 * unchecked exception here is because the caller cannot reasonably
+			 * be expected to recover from an exception. Cf
+			 * https://docs.oracle.com/javase/tutorial/essential/exceptions/
+			 * runtime.html
+			 * =================================================================
+			 * ======================================
+			 */
+			throw new RuntimeException(e);
+		}
+	}
+
+	/** Prints the Goodbye message and exits. */
+	private void exit() {
+		ui.showGoodbyeMessage();
+		System.exit(0);
+	}
+
+	/**
+	 * Reads the user command and executes it, until the user issues the exit
+	 * command.
+	 */
+	private void runCommandLoopUntilExitCommand() {
+		try {
+			Command command;
+			do {
+				String userCommandText = ui.getUserCommand();
+				command = new Parser().parseCommand(userCommandText);
+				CommandResult result = executeCommand(command);
+				recordResult(result);
+				ui.showResultToUser(result);
+
+			} while (!ExitCommand.isExit(command));
+		} catch (FileDeletedException e) {
+			ui.showInitFailedMessage();
+			throw new RuntimeException(e);
+		}
+	}
+
+	/**
+	 * Updates the {@link #lastShownList} if the result contains a list of
+	 * Persons.
+	 */
+	private void recordResult(CommandResult result) {
+		final Optional<List<? extends ReadOnlyPerson>> personList = result.getRelevantPersons();
+		if (personList.isPresent()) {
+			lastShownList = personList.get();
+		}
+	}
+
+	/**
+	 * Executes the command and returns the result.
+	 * 
+	 * @param command
+	 *            user command
+	 * @return result of the command
+	 */
+	private CommandResult executeCommand(Command command) throws FileDeletedException {
```

Comment:
It is usually not recommended to catch an exception, and throw another. You may actually just warn the user or terminate the program. Recall from the tutorial discussion which would be a better choice.

In this program, RunTimeException is not handled. This means that the program could not recover from the situation where the storage file is deleted and crash. Is this what you have planned for the program to do?